File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.py
class Solution(object):
    def minCostClimbingStairs(self, cost):
        """
        :type cost: List[int]
        :rtype: int
        """
        cost.append(0)

        for i in range(2, len(cost)):
            cost[i] += min(cost[i-1], cost[i-2])
        
        return cost[-1]
        

File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.go
func minCostClimbingStairs(cost []int) int {
    n := len(cost)
    ans := make([]int, n)
    ans[0] = cost[0]
    ans[1] = cost[1]
    for i := 2; i < n; i++ {
        ans[i] = min(ans[i-1], ans[i-2]) + cost[i]
    }

    return min(ans[n-1], ans[n-2])
}